src/library_addfunction.js

```diff
@@ -44,59 +44,62 @@ mergeInto(LibraryManager.library, {
     }
     return type;
   },
-
-  // Wraps a JS function as a wasm function with a given signature.
-  $convertJsFunctionToWasm__deps: ['$uleb128Encode', '$sigToWasmTypes'],
-  $convertJsFunctionToWasm: function(func, sig) {
-#if WASM2JS
-    return func;
-#else // WASM2JS
-
-    // If the type reflection proposal is available, use the new
-    // "WebAssembly.Function" constructor.
-    // Otherwise, construct a minimal wasm module importing the JS function and
-    // re-exporting it.
-    if (typeof WebAssembly.Function == "function") {
-      return new WebAssembly.Function(sigToWasmTypes(sig), func);
-    }
-
-    // The module is static, with the exception of the type section, which is
-    // generated based on the signature passed in.
-    var typeSectionBody = [
-      0x01, // count: 1
-      0x60, // form: func
-    ];
+  $generateFuncType__deps: ['$uleb128Encode'],
+  $generateFuncType : function(sig, target){
     var sigRet = sig.slice(0, 1);
     var sigParam = sig.slice(1);
     var typeCodes = {
       'i': 0x7f, // i32
-#if MEMORY64
+  #if MEMORY64
```

Comment:
We tend not to indent our preprocessor statement.    Can you keep all of these in column 0?